integration/integration-uiSchema.spec.ts

```diff
@@ -0,0 +1,129 @@
+
+  // Expect uiSchema to be working, i.e. it renders the remarks field to be a textarea
+  await t
+    .expect(Form.find("textarea#root_remarks").value)
+    .contains("Some very important documents in here for some submission");
+
+  // Add new form
+  await t.click(AddNewButton);
+
+  /*
+   * This test case is here in the event that users really
+   * want to have a `uiSchema` attribute present in their
+   * TradeTrust document.
+   */
+
+  // Navigate to form
+  await t.click(Button.withText("Covering Letter (DBS, Nested UISchema)"));
+  await t.expect(Title.textContent).contains("Fill and Preview Form");
+  await t.expect(ProgressBar.textContent).contains("Step 2/3");
+
+  // Validate that default values is populated
+  await t.expect(DocumentTitleField.value).contains("Documents Bundle");
+  await t
+    .expect(DocumentRemarksField.value)
+    .contains("Some very important documents in here for some submission");
+
+  // Expect remarks field to be a normal input field (because uiSchema does not exist at the form template's root level)
+  await t
+    .expect(Form.find("input#root_remarks").value)
+    .contains("Some very important documents in here for some submission");
+
+  // Submit
+  await t.click(SubmitButton);
+
+  // pending confirmation of issued documents
+  await t.expect(Selector("[data-testid='publish-loader']").exists).ok();
+
+  // Check that the two Covering Letters are created
+  const fileName1 = "Document-1-local.tt";
+  const fileName2 = "Document-2-local.tt";
+  await t.expect(Title.textContent).contains("Document(s) issued successfully");
+  await t.expect(Selector("div").withText(fileName1).exists).ok();
+  await t.expect(Selector("div").withText(fileName2).exists).ok();
+  await t.expect(Selector("div").withText("Download").exists).ok();
+
+  // Check that the first Covering Letter has been downloaded successfully
+  await t.click(downloadLink.nth(0));
+  const filePath1 = `${downloadsFolder()}/${fileName1}`;
+  await t.expect(await waitForFileDownload(t, filePath1)).eql(true);
+
+  // We expect the contents of the first doc to NOT contain uiSchema key
+  // Only doing a `contains` check because the value is non-deterministic due to the hash
+  const docContents1 = JSON.parse(readFileSync(filePath1, "utf8"));
+  console.log(docContents1);
+  await t.expect(docContents1.data).notContains({ uiSchema: { remarks: { "ui:widget": "" } } });
+
+  // Check that the second Covering Letter has been downloaded successfully
+  await t.click(downloadLink.nth(1));
+  const filePath2 = `${downloadsFolder()}/${fileName2}`;
+  await t.expect(await waitForFileDownload(t, filePath2)).eql(true);
+
+  // We expect the contents of the second doc to contain the uiSchema key,
+  // one at the root level, the other nested in `misc`
+  const docContents2 = readFileSync(filePath2, "utf8");
+  console.log(docContents2);
+  await t.expect(docContents2).contains("uiSchema");
```

Comment:
I couldn't figure out how to use Testcafe to search for specific JSON nodes.

My idea was to expect `docContents2` to contain two `uiSchema` keys. I wanted to check only the keys, so I can't really use `.contains( { "uiSchema": { "remarks": { "ui:widget": "RANDOM_HASH:string:textarea" } } })` because the `RANDOM_HASH` is non-deterministic.

Instead, I am just checking for the presence of only one `uiSchema`, `remarks`, etc. as the next best alternative.

Reply:
So instead of parsing `docContents2` as JSON, I am just leaving it as-is, so that `contains` can just search through a text version of the TradeTrust document.

For reference,
`{"version":"https://schema.openattestation.com/2.0/schema.json","data":{"$template":{"type":"b9423e56-d747-4683-a9ba-7e182bbd1b8c:string:EMBEDDED_RENDERER","name":"5f1de081-7364-4c40-8d33-299fccba3486:string:COVERING_LETTER","url":"c1eaa840-36da-49fe-8065-565dec493d21:string:https://generic-templates.tradetrust.io"},"issuers":[{"name":"a416d63b-313b-4ad9-9d0f-4095b9be04c7:string:Demo Issuer","documentStore":"b1ae58d5-8862-4f95-8516-83da21512f3a:string:0x63a223e025256790e88778a01f480eba77731d04","identityProof":{"type":"73a2aeb1-b8b1-458a-b5fd-90129bd52e2c:string:DNS-TXT","location":"9bbdaa0f-5965-4c7b-8e05-df46590c9d0e:string:demo-tradetrust.openattestation.com"}}],"name":"fa9f5a51-5e58-4b01-a29b-a807610d0b0e:string:Covering Letter","logo":"dee7fbeb-30b2-47d6-bc01-029c228d3410:string:https://pluspng.com/img-png/logo-dbs-png-file-dbs-bank-logo-svg-1280.png","backgroundColor":"67930a3f-fb7d-41f0-947f-3793aef1c149:string:black","titleColor":"63f488e0-2dec-42bf-a9eb-55e4ffd83b22:string:white","remarksColor":"ab9ee80c-4539-4cf0-8e9b-409ad17f1a20:string:white","title":"c8617874-2585-49bc-88d8-e35bb5506a1d:string:Documents Bundle","remarks":"1140c5ab-633e-43e0-9101-3f56cd6d8e7f:string:Some very important documents in here for some submission","misc":{"miscKey1":"ee320769-5843-4411-bcb8-9a83f170fb95:string:miscValue1","uiSchema":{"remarks":{"ui:widget":"4a7fa1d1-a096-4da6-8957-a8f65809bf06:string:textarea"}}},"uiSchema":{"remarks":{"ui:widget":"c003dbdc-2268-4518-a41c-a4a4e56fa6f8:string:textarea"}}},"signature":{"type":"SHA3MerkleProof","targetHash":"3f70323bd6a8bf9f8292e07c01f719156cbc7150c079beb018f524cc67eb5290","proof":["1df26567040c975ccd359137e362cb97e46f6a598d136dc2fc767b3e25d9af84"],"merkleRoot":"f44c0c7ef43f1e947ec159c34ff5d11c1c0804f140bdfb9b90ebb8ed8d420629"}}`

Reply:
![image](https://user-images.githubusercontent.com/7406870/97253765-f3629180-1847-11eb-9cb7-b2a19abe654d.png)

Reply:
Thank you for this!